src/converse-autocomplete.js

```diff
@@ -22,13 +22,24 @@ export const FILTER_STARTSWITH = function (text, input) {
 };
 
 
-const SORT_BYLENGTH = function (a, b) {
+const SORT_BY_LENGTH = function (a, b) {
     if (a.length !== b.length) {
         return a.length - b.length;
     }
     return a < b? -1 : 1;
 };
 
+const SORT_BY_QUERY_POSITION = function (a, b) {
+    const query = a.query;
+    const x = a.label.toLowerCase().indexOf(query);
+    const y = b.label.toLowerCase().indexOf(query);
+
+    if (x === y) {
+        return SORT_BY_LENGTH(a, b);
+    }
+    return (x === -1 ? Infinity : x) < (y === -1 ? Infinity : y) ? -1 : 1
```

Comment:
Shouldn't `x` and `y` never be `-1`?

If this sorting function is called for `x` and `y`, then it means they already match the query string, so `indexOf` should never return `-1`. Or is there something I'm missing?

Reply:
Sometimes this happens when a "list" is directly passed to an instance of autocomplete as in the following case: 

https://github.com/conversejs/converse.js/blob/5ea2ccfc17ebb0360dcdde80321cb2e1a8b1fc20/src/modals/muc-invite.js#L31-L35

I don't know why, but checking that the index is different to -1, allowed me to correct this.

Reply:
Ok, then I think your change makes sense 👍